contrib/vagrant/start.sh

```diff
@@ -576,36 +576,92 @@ function vboxnet_addr_finder(){
     vboxnet_add_ipv4 "${vboxnetname}" "${IPV4_BASE_ADDR_NFS}" "255.255.255.0"
 }
 
-vboxnet_addr_finder
-
-ipv6_public_workers_addrs=()
-
-split_ipv4 ipv4_array "${MASTER_IPV4}"
-MASTER_IPV6="${IPV6_INTERNAL_CIDR}$(printf '%02X' ${ipv4_array[3]})"
-
-set_reload_if_vm_exists
-
-create_master
-create_workers
-set_vagrant_env
-create_k8s_config
-
-cd "${dir}/../.."
+#Code for Detecting and warning user
+
+function createVm(){
+	vboxnet_addr_finder
+
+	ipv6_public_workers_addrs=()
+
+	split_ipv4 ipv4_array "${MASTER_IPV4}"
+	MASTER_IPV6="${IPV6_INTERNAL_CIDR}$(printf '%02X' ${ipv4_array[3]})"
+
+	set_reload_if_vm_exists
+
+	create_master
+	create_workers
+	set_vagrant_env
+	create_k8s_config
+
+	cd "${dir}/../.."
+
+	PROVISION_ARGS=""
+	if [ -n "${NO_PROVISION}" ]; then
+		PROVISION_ARGS="--no-provision"
+	fi
+	if [ -n "${RELOAD}" ]; then
+		vagrant reload $PROVISION_ARGS $1
+	elif [ -n "${PROVISION}" ]; then
+		vagrant provision $1
+	else
+		vagrant up $PROVISION_ARGS $1
+		if [ "$?" -eq "0" -a -n "${K8S}" ]; then
+			host_port=$(vagrant port --guest 6443 k8s1)
+			vagrant ssh k8s1 -- cat /home/vagrant/.kube/config | sed "s;server:.*:6443;server: https://k8s1:$host_port;g" > vagrant.kubeconfig
+			echo "Add '127.0.0.1 k8s1' to your /etc/hosts to use vagrant.kubeconfig file for kubectl"
+		fi
+	fi
+}
 
-PROVISION_ARGS=""
-if [ -n "${NO_PROVISION}" ]; then
-    PROVISION_ARGS="--no-provision"
-fi
-if [ -n "${RELOAD}" ]; then
-    vagrant reload $PROVISION_ARGS $1
-elif [ -n "${PROVISION}" ]; then
-    vagrant provision $1
+runningVm=$(VBoxManage list runningvms | awk 'END{ print NR }')
+VMName=$(VBoxManage list runningvms | awk 'NR==1{print $1}' |  cut -d "\"" -f 2)
+if [ "$VMName" ]; then
+	echo
+	echo "Detected running VMs that might cause conflict"
+	echo
+	VBoxManage list runningvms
+	echo
+	printf "Do you wish to pause (p/P), stop (s/S), or destroy (d/D)?  "
+	read optn
+	case "$optn" in
+		"p" | "P")
+			for ((i=1; i<=$runningVm; i=i+1))
+			do
+				VMName=$(VBoxManage list runningvms | awk 'NR=='$i'{print $1}' |  cut -d "\"" -f 2)
+				echo
+				echo $VMName
+				VBoxManage controlvm $VMName pause
+			done
+			printf "\n$runningVm VM(s) paused successfully\n" 
+			printf "Tip: Try running vagrant status to check status\n"
```

Comment:
Pausing the VM is not currently a viable option because the script doesn't seem to distinguish between paused and running VMs:
```
$ NO_BUILD=1 K8S=1 ./contrib/vagrant/start.sh 

Detected running VMs that might cause conflict

"cilium3_k8s1_1609951352422_54263" {93a231c6-c3f9-4366-a20f-35f60623b33b}

Do you wish to pause (p/P), stop (s/S), or destroy (d/D)?  p

cilium3_k8s1_1609951352422_54263

1 VM(s) paused successfully
Tip: Try running vagrant status to check status
$
$
$ NO_BUILD=1 K8S=1 ./contrib/vagrant/start.sh 

Detected running VMs that might cause conflict

"cilium3_k8s1_1609951352422_54263" {93a231c6-c3f9-4366-a20f-35f60623b33b}

Do you wish to pause (p/P), stop (s/S), or destroy (d/D)?  p

cilium3_k8s1_1609951352422_54263
VBoxManage: error: Already paused
VBoxManage: error: Details: code VBOX_E_INVALID_VM_STATE (0x80bb0002), component ConsoleWrap, interface IConsole, callee nsISupports
VBoxManage: error: Context: "Pause()" at line 405 of file VBoxManageControlVM.cpp

1 VM(s) paused successfully
Tip: Try running vagrant status to check status
```

Reply:
` $NO_BUILD=1 K8S=1 ./contrib/vagrant/start.sh

Detected running VMs that might cause conflict

"cilium_k8s1_1610031382439_78313" {a3c160c3-9f2b-4aae-8c87-44e004a7a003}

Do you wish to pause, stop, or destroy the VM(s) or ignore and continue? [p/s/d/C]  p

cilium_k8s1_1610031382439_78313 paused

1 VM(s) paused successfully
Tip: Try running vagrant status to check status
$
$ NO_BUILD=1 K8S=1 ./contrib/vagrant/start.sh

Detected running VMs that might cause conflict

"cilium_k8s1_1610031382439_78313" {a3c160c3-9f2b-4aae-8c87-44e004a7a003}

Do you wish to pause, stop, or destroy the VM(s) or ignore and continue? [p/s/d/C]  p

cilium_k8s1_1610031382439_78313 already in paused state

0 VM(s) paused successfully
Tip: Try running vagrant status to check status
`